src/Compilers/CSharp/Portable/BoundTree/BoundDagTest.cs

```diff
@@ -21,6 +21,8 @@ private bool Equals(BoundDagTest? other)
                 return false;
             if (this == other)
                 return true;
+            if (!this.Input.Equals(other.Input))
+                return false;
```

Comment:
:memo: This hasn't been checked before, probably because it was guaranteed to be true, but now that we're rewriting tests with a new input, this could be actually false.